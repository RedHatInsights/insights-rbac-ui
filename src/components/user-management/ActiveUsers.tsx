import { useChrome } from '@redhat-cloud-services/frontend-components/useChrome';
import { useFlag } from '@unleash/proxy-client-react';
import React, { FunctionComponent, useContext } from 'react';
import PermissionsContext, { PermissionsContextType } from '../../utilities/permissionsContext';
import { ActiveUsersAdminView } from './ActiveUsersAdminView';
import { ActiveUsersNonAdminView } from './ActiveUsersNonAdminView';

type ActiveUserProps = {
  linkDescription?: string;
  linkTitle?: string;
};

export const ActiveUsers: FunctionComponent<ActiveUserProps> = ({ linkDescription, linkTitle }) => {
  const chrome = useChrome();
  const env = chrome.getEnvironment();
  const prefix = chrome.isProd() ? '' : `${env}.`;
  const { orgAdmin } = useContext(PermissionsContext) as PermissionsContextType;
  const isITLess = useFlag('platform.rbac.itless');
  return !isITLess && orgAdmin ? (
    <ActiveUsersAdminView linkDescription={linkDescription} linkTitle={linkTitle} prefix={prefix} />
  ) : (
    <ActiveUsersNonAdminView />
  );
};
