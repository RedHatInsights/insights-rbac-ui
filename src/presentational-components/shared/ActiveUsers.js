import React from 'react';
import PropTypes from 'prop-types';
import { TextContent, Text, TextVariants } from '@patternfly/react-core';
const ActiveUser = ({ description, linkTitle }) => (
  <TextContent>
    <Text className="pf-u-mt-0" component={TextVariants.h7}>
      {description}
      <Text
        component={TextVariants.a}
        href={`https://www.${window.insights.chrome.isProd ? '' : 'qa.'}redhat.com/wapps/ugc/protected/usermgt/userList.html`}
        target="_blank"
        rel="noopener noreferrer"
      >
        {linkTitle}
      </Text>
    </Text>
  </TextContent>
);

ActiveUser.propTypes = {
  description: PropTypes.node,
  linkTitle: PropTypes.node,
};

ActiveUser.defaultProps = {
  description: '',
  linkTitle: ' user management list.',
};

export default ActiveUser;
