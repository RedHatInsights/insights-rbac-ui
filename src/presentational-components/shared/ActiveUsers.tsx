import { useChrome } from '@redhat-cloud-services/frontend-components/useChrome';
import { useFlag } from '@unleash/proxy-client-react';
import React, { FunctionComponent, useContext } from 'react';
import PermissionsContext from '../../utilities/permissions-context';
import { ActiveUsersAdminView } from './ActiveUsersAdminView';
import { ActiveUsersNonAdminView } from './ActiveUsersNonAdminView';

type ActiveUserProps = {
  linkDescription?: string;
  linkTitle?: string;
};

const ActiveUser: FunctionComponent<ActiveUserProps> = ({ linkDescription, linkTitle }) => {
  const chrome = useChrome();
  const env = chrome.getEnvironment();
  const prefix = chrome.isProd() ? '' : `${env}.`;
  const { orgAdmin } = useContext(PermissionsContext);
  const isITLess = useFlag('platform.rbac.itless');
  return !isITLess && orgAdmin ? (
    <ActiveUsersAdminView linkDescription={linkDescription} linkTitle={linkTitle} prefix={prefix} />
  ) : (
    <ActiveUsersNonAdminView />
  );
};

export default ActiveUser;
