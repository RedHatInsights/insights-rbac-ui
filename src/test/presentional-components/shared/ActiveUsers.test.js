import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ActiveUsers from '../../../presentational-components/shared/ActiveUsers';

describe('<ActiveUsers />', () => {
  it('should render correctly', () => {
    expect(toJson(shallow(<ActiveUsers />))).toMatchSnapshot();
  });

  it('should render correctly with description', () => {
    expect(toJson(shallow(<ActiveUsers description="some description" />))).toMatchSnapshot();
  });

  it('should render correctly with link title', () => {
    expect(toJson(shallow(<ActiveUsers linkTitle="some title" />))).toMatchSnapshot();
  });
});
