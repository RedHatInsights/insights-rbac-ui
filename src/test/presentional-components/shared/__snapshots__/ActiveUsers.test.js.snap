// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveUsers /> should render correctly 1`] = `
<TextContent>
  <Text
    className="pf-u-mt-0"
  >
    <Text
      component="a"
      href="https://www.qa.redhat.com/wapps/ugc/protected/usermgt/userList.html"
      rel="noopener noreferrer"
      target="_blank"
    >
       user management list.
    </Text>
  </Text>
</TextContent>
`;

exports[`<ActiveUsers /> should render correctly with description 1`] = `
<TextContent>
  <Text
    className="pf-u-mt-0"
  >
    some description
    <Text
      component="a"
      href="https://www.qa.redhat.com/wapps/ugc/protected/usermgt/userList.html"
      rel="noopener noreferrer"
      target="_blank"
    >
       user management list.
    </Text>
  </Text>
</TextContent>
`;

exports[`<ActiveUsers /> should render correctly with link title 1`] = `
<TextContent>
  <Text
    className="pf-u-mt-0"
  >
    <Text
      component="a"
      href="https://www.qa.redhat.com/wapps/ugc/protected/usermgt/userList.html"
      rel="noopener noreferrer"
      target="_blank"
    >
      some title
    </Text>
  </Text>
</TextContent>
`;
